src/js/common/components/Search/SearchBar2024.jsx

```diff
@@ -56,12 +58,30 @@ class SearchBar2024 extends Component {
     }
   }
 
-  handleKeyPress () {
-    if (this.timer) clearTimeout(this.timer);
-    this.timer = setTimeout(() => {
-      this.props.searchFunction(this.state.searchString);
-    }, this.props.searchUpdateDelayTime);
+  
+ handleKeyPress = () => {
+  if (this.timer) {
+    clearTimeout(this.timer);
   }
+  this.timer = setTimeout(() => {
+    const { searchString } = this.state;
+    if (searchString.length === 0) {
+      return;
+    }
+    this.props.searchFunction(searchString);
+    if(this.props.trackSearch){
+    const dataLayerObject = {
+      event: 'KeyWord Search',
+      search_keyword: searchString,
```

Comment:
I believe 'search_keyword' needs to be camelCase like: 'searchKeyword'. Also, I added searchKeyword to our list of Property Names. https://docs.google.com/spreadsheets/d/1KO4_AOmddKvIpA79XrZdaVBdRkcP1HitqrvQkCo2A34/edit?gid=1828322090#gid=1828322090